include/hffix.hpp

```diff
@@ -2314,16 +2318,27 @@ class message_reader {
             return;
         }
 
+        // Added check for expected CheckSum tag '10='
+        if (*checksum != '1' || checksum[1] != '0' || checksum[2] != '=') {
+            invalid();
+            return;
+        }           
+          
         begin_.buffer_ = b;
         begin_.current_.tag_ = 35; // MsgType
         b += 3;
         begin_.current_.value_.begin_ = b;
-        while(*++b != '\x01') {
-            if (b >= checksum) {
-                invalid();
-                return;
+  
+        // Handle case of MsgType value could be empty
```

Comment:
The purpose of this loop is to make sure that the `MsgType` is terminated somewhere before the end of the message by `\x01`. This is just so that we are guaranteed to not read off the end of the buffer when we try to read the `MsgType`.

I think your code is an improvement, because if the `MsgType` is empty and then the buffer ends immediately after the `0x01`, then hffix might read off the end of the buffer. Would you please refactor this PR so that this check is in a commit by itself?

Could you also please provide a test case which fails on the old code when the buffer ends right after `35=\x01`?